src/config.rs

```diff
@@ -181,10 +200,19 @@ impl Secrets {
             EncryptionClient::new(client)
         }
 
-        #[cfg(not(feature = "aws"))]
+        #[cfg(feature = "aes")]
         {
             let client = self.master_key;
             EncryptionClient::new(client)
         }
+
+        // TODO: Add Vault instance
+        #[cfg(feature = "vault")]
+        {
+            // This function doesn't return result therefore unwrapping.
+            // TODO: Make it customresult return type
+            let client = init_vault(self.vault_config, self.vault_token).unwrap();
```

Comment:
The panic part can happen inside the `init_vault` function

Reply:
I see there are many expect methods used through out the code that can also panic. Rewired unwrap to expect.